src/main/java/picard/cmdline/CommandLineProgram.java

```diff
@@ -329,7 +329,7 @@ public int instanceMain(final String[] argv) {
     */
     protected boolean parseArgs(final String[] argv) {
 
-        commandLineParser = getCommandLineParser();
+        commandLineParser = determineArgParser(argv);
```

Comment:
The code here is a bit sloppy with regard to when `commandLineParser` is initialized. It looks like there are a few different call chains where `getCommandLineParser()` is called, so it's not immediately obvious to me that this line will occur before any other call to `getCommandLineParser()` occurs. It would be much better if we had control over the first place where this is initialized and used the field directly in all other places, to make it clear that this initialization must occur first.

Reply:
Per @pshapiro4broad comment above, this change should be integrated into `getCommandLineParser`.